control/controlexecute/control_run_lifecycle.go

```diff
@@ -0,0 +1,72 @@
+package controlexecute
+
+import "time"
+
+type ControlRunLifecycleEvent string
+
+const (
+	ControlRunLifecycleEventConstructed           ControlRunLifecycleEvent = "constructed"
```

Comment:
I still feel this could be lighter still - just call `lifecycle.Add("execute_start")` rather than define multiple functions

Reply:
then lifecycle becomes very minimal - it is only gather log informations o the magic strings do not worry me (for once)

And if we ever want to gather additional information we do not need to add another function